mlflow/entities/assessment.py

```diff
@@ -0,0 +1,187 @@
+from typing import Any, Dict, Optional
+
+from mlflow.entities._mlflow_object import _MlflowObject
+from mlflow.entities.assessment_source import AssessmentSource
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+
+
+class Assessment(_MlflowObject):
+    """
+    Assessment data associated with an evaluation.
+    """
+
+    def __init__(
+        self,
+        evaluation_id: str,
+        name: str,
+        source: AssessmentSource,
+        timestamp: int,
+        boolean_value: Optional[bool] = None,
+        numeric_value: Optional[float] = None,
+        string_value: Optional[str] = None,
+        rationale: Optional[str] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+        error_code: Optional[str] = None,
+        error_message: Optional[str] = None,
+    ):
+        """Construct a new mlflow.entities.Assessment instance.
+
+        Args:
+            evaluation_id: The ID of the evaluation with which the assessment is associated.
+            name: The name of the assessment.
+            source: The source of the assessment (AssessmentSource instance).
+            timestamp: The timestamp when the assessment was given.
+            boolean_value: The boolean assessment value, if applicable.
+            numeric_value: The numeric assessment value, if applicable.
+            string_value: The string assessment value, if applicable.
+            rationale: The rationale / justification for the value.
+            metadata: Additional metadata for the assessment, e.g. the index of the chunk in the
+                      retrieved documents that the assessment applies to.
+            error_code: An error code representing any issues encountered during the assessment.
+            error_message: A descriptive error message representing any issues encountered during
+                the assessment.
+        """
+        self._evaluation_id = evaluation_id
+        self._name = name
+        self._source = source
+        self._timestamp = timestamp
+        self._boolean_value = boolean_value
+        self._numeric_value = numeric_value
+        self._string_value = string_value
+        self._rationale = rationale
+        self._metadata = metadata or {}
+        self._error_code = error_code
+        self._error_message = error_message
+
+        if error_message and any(
+            [boolean_value is not None, numeric_value is not None, string_value is not None]
+        ):
+            raise MlflowException(
+                "error_message cannot be specified when boolean_value, numeric_value, "
+                "or string_value is specified.",
+                INVALID_PARAMETER_VALUE,
+            )
```

Comment:
```suggestion
        if error_message and(
            boolean_value is not None or numeric_value is not None or string_value is not None
        ):
            raise MlflowException(
                "error_message cannot be specified when boolean_value, numeric_value, "
                "or string_value is specified.",
                INVALID_PARAMETER_VALUE,
            )
```

nit: to avoid useless evaluation.

Reply:
Fixed!